### PR TITLE
Initial support for time stamps

### PR DESCRIPTION
**Pull request by kjetil-kilhavn**
Cells which are formatted with both date format codes and time format codes are converted to time stamp values instead of date values. There is no support for time zones - all values will be converted to UTC time stamps.

This solves #1130 and includes the fix for #1132 also.
Additionally there was a bug in the code which determined field conversion exits per column. Data from the previous loop iteration (previous column) was used for non-zero sy-subrc.

### CODE REVIEW

File: src/zcl_excel_worksheet.clas.abap [hunk 1]
```diff
@@ -2265,8 +2276,13 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
         REPLACE ALL OCCURRENCES OF REGEX '\[\L[^]]*\]' IN lv_format_code WITH ''.
 
         IF lv_format_code CA 'yd' OR lv_format_code EQ zcl_excel_style_number_format=>c_format_date_std.
-          " DATE = yyyymmdd
-          ls_style_conv-abap_type = cl_abap_typedescr=>typekind_date.
+          "TODO kjetil-kilhavn make use of is_date_format( ) and is_time_format( )
```
Comment (sandraros) [inline src/zcl_excel_worksheet.clas.abap:2279]: Could you finish the "TODO"?

File: src/zcl_excel_worksheet.clas.abap [hunk 2]
```diff
@@ -3273,6 +3291,31 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
   ENDMETHOD.                    "IS_CELL_MERGED
 
 
+  METHOD is_date_format.
+    "TODO kjetil-kilhavn figure out how to detect all valid date formats, including mmmmm (first letter of month name)
```
Comment (sandraros) [inline src/zcl_excel_worksheet.clas.abap:3295]: Could you finish the "TODO"?

File: src/zcl_excel_worksheet.clas.abap [hunk 3]
```diff
@@ -3273,6 +3291,31 @@ CLASS zcl_excel_worksheet IMPLEMENTATION.
...
+    result = abap_false.
+
+    CHECK ip_value CA 'ymd'.
+
+    result = abap_true.
+  ENDMETHOD. "is_date_format
+
+
+  METHOD is_time_format.
+    "TODO kjetil-kilhavn try to implement this without hardcoding too much...
```
Comment (sandraros) [inline src/zcl_excel_worksheet.clas.abap:3305]: Could you finish the "TODO"?

**Review by sandraros — COMMENTED** [review]
Sorry for the long delay. I see in your proposed code several "TODO", could you finish that part? Thank you.

**Comment by kjetil-kilhavn** [comment]
> Sorry for the long delay. I see in your proposed code several "TODO", could you finish that part? Thank you.

I will try, but it will take time as it is no longer work-related and the standards document isn't light reading. I suppose there's no rush though since there apparently hasn't been a huge demand for this.

I'll split off the error correction into a separate pull request so that can be merged first. Hopefully I can get it done this week.

**Comment by kjetil-kilhavn** [comment]
I created pull request #1197 to ensure the date value in ABAP is always the same as the date displayed in the spreadsheet. A runtime error for time conversion is also fixed (ignore integer part of value) in the same pull request.

The timestamp issue, and the suggested refactoring to methods for date and time formats is a big task, and I think it is better to just close this request without merging now.

**Comment by kjetil-kilhavn** [comment]
Closing.

**Pull request closed without merging**